forjj secrets edit segfault

This error happens on a path which should not exist, but stay visible. The error happen when we set a value to this secret path.

```bash
$ forjj secrets edit app/jenkins/aws-iam-arn-slave
branch HEAD, build_date 2018-07-28_23:09:14_+0200, build_commit c79be8b102c090bfa12a76e6157273bc6074345e, build_tag false.
WARNING ! forjj/forjfile.(*Forge).Load: Both, a Forjfile and a Forjfile model were loaded. The model has been ignored.
2018/07/31 11:23:29 Organization : 'pdxc-software-factory'
WARNING ! main.(*Forj).init_driver_flags: No actions to add flags.
WARNING ! main.(*Forj).init_driver_flags: No actions to add flags.
WARNING ! main.(*Forj).init_driver_flags: No actions to add flags.
WARNING ! forjj/vendor/github.com/forj-oss/forjj-modules/cli.(*ForjObject).AddInstanceField: Field pro-deployment already added in app-jenkins. Ignored.
2018/07/31 11:23:31 Set default upstream application to 'github'
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x75ef42]

goroutine 1 [running]:
forjj/vendor/github.com/forj-oss/goforjj.(*ValueStruct).Equal(0xc42015bdf8, 0x0, 0x7ffef60198fc)
        /go/workspace/src/forjj/vendor/github.com/forj-oss/goforjj/value.go:162 +0x32
forjj/creds.(*yamlSecure).setObjectValue(0xc4201af4a0, 0x8fb4a3, 0x5, 0x7ffef60198f0, 0x3, 0x7ffef60198f4, 0x7, 0x7ffef60198fc, 0x11, 0xc42015bdf8, ...)
        /go/workspace/src/forjj/creds/yaml_secure.go:174 +0x147
forjj/creds.(*Secure).SetObjectValue(0xb89af0, 0xc420164ab8, 0x3, 0x8fb4a3, 0x5, 0x7ffef60198f0, 0x3, 0x7ffef60198f4, 0x7, 0x7ffef60198fc, ...)
        /go/workspace/src/forjj/creds/creds.go:284 +0x105
main.(*secretsEdit).doEdit(0xb89888)
        /go/workspace/src/forjj/secrets_edit.go:124 +0x8dd
main.(*secrets).action(0xb897f0, 0xc420429cf4, 0xc)
        /go/workspace/src/forjj/secrets.go:54 +0xb8
main.(*secrets).(main.action)-fm(0xc420429cf4, 0xc)
        /go/workspace/src/forjj/app.go:243 +0x3e
main.main()
        /go/workspace/src/forjj/forjj.go:52 +0x1ed

```

--- a/creds/yaml_secure.go
+++ b/creds/yaml_secure.go
@@ -51,7 +51,7 @@ func (d *yamlSecure) foundFiles() (ret []string) {
 func (d *yamlSecure) load(env string, secretFile bool) error {
 	d.foundFiles()
 	file := d.files[0]
-	if ! secretFile {
+	if !secretFile {
 		file = d.files[1]
 	}
 	if file == "" {
@@ -93,7 +93,7 @@ func (d *yamlSecure) save(secretFile bool) (err error) {
 		yamlData []byte
 	)
 	file := d.credFile
-	if ! secretFile {
+	if !secretFile {
 		file = d.file
 		yamlData, err = yaml.Marshal(d)
 	} else {
@@ -170,7 +170,7 @@ func (d *yamlSecure) setObjectValue(source, obj_name, instance_name, key_name st
 		keys[key_name] = newValue
 		d.Objects[obj_name][instance_name] = keys
 		updated = true
-	} else if v, found := k[key_name]; found {
+	} else if v, found := k[key_name]; found && v != nil {
 		if !value.Equal(v) {
 			*v = *value
 			updated = true

--- a/glide.lock
+++ b/glide.lock
@@ -1,5 +1,5 @@
 hash: 1e7781913786fccab1c3c054ef35e4e9b0608ce6b13f46ef26ae9210212f496c
-updated: 2018-07-20T16:12:03.52471137Z
+updated: 2018-07-31T20:29:02.820953607Z
 imports:
 - name: github.com/alecthomas/kingpin
   version: a328427ab7d619fe3c8d16a0da66899d03d5afae
@@ -21,7 +21,7 @@ imports:
   - cli/tools
   - trace
 - name: github.com/forj-oss/goforjj
-  version: e32c7a7dc655c391d5f9a4783ac1236e3311cc08
+  version: 9f28388b6e9cdbd799c14f8a24527031c2e845b8
 - name: github.com/kr/text
   version: e2ffdb16a802fe2bb95e2e35ff34f0e53aeef34f
 - name: github.com/kvz/logstreamer
@@ -37,19 +37,22 @@ imports:
 - name: github.com/pkg/errors
   version: 816c9085562cd7ee03e7f8188a1cfd942858cded
 - name: golang.org/x/crypto
-  version: a2144134853fc9a27a7b1e3eb4f19f1a76df13c9
+  version: c126467f60eb25f8f27e5a981f32a87e3965053f
   subpackages:
   - ssh/terminal
 - name: golang.org/x/net
-  version: a680a1efc54dd51c040b3b5ce4939ea3cf2ea0d1
+  version: 49c15d80dfbc983ea25246ee959d970efe09ec09
   subpackages:
+  - internal/socks
   - proxy
   - publicsuffix
 - name: golang.org/x/sys
-  version: ac767d655b305d4e9612f5f6e33120b9176c4ad4
+  version: bd9dbc187b6e1dacfdd2722a87e83093c2d7bd6e
   subpackages:
   - unix
   - windows
+- name: gopkg.in/alecthomas/kingpin.v2
+  version: 947dcec5ba9c011838740e680966fd7087a71d0d
 - name: gopkg.in/yaml.v2
   version: 5420a8b6744d3b0345ab293f6fcba19c978f1183
 testImports: []

--- a/glide.lock
+++ b/glide.lock
@@ -1,5 +1,5 @@
 hash: 1e7781913786fccab1c3c054ef35e4e9b0608ce6b13f46ef26ae9210212f496c
-updated: 2018-07-31T20:29:02.820953607Z
+updated: 2018-07-31T21:09:08.048568769Z
 imports:
 - name: github.com/alecthomas/kingpin
   version: a328427ab7d619fe3c8d16a0da66899d03d5afae
@@ -21,7 +21,7 @@ imports:
   - cli/tools
   - trace
 - name: github.com/forj-oss/goforjj
-  version: 9f28388b6e9cdbd799c14f8a24527031c2e845b8
+  version: 75c9f7b93feda0cb351fb5a2b810d65bfefb1f4a
 - name: github.com/kr/text
   version: e2ffdb16a802fe2bb95e2e35ff34f0e53aeef34f
 - name: github.com/kvz/logstreamer

--- a/secrets_edit.go
+++ b/secrets_edit.go
@@ -16,11 +16,12 @@ import (
 )
 
 type secretsEdit struct {
-	editor   *string
-	cmd      *kingpin.CmdClause
-	key      *string
-	password string
-	common   *secretsCommon
+	editor          *string
+	noTerminalSetup *bool
+	cmd             *kingpin.CmdClause
+	key             *string
+	password        string
+	common          *secretsCommon
 
 	elements map[string]secretInfo
 }
@@ -30,6 +31,7 @@ func (s *secretsEdit) init(parent *kingpin.CmdClause, common *secretsCommon) {
 	s.cmd = parent.Command("edit", "edit a credential stored in forjj secrets")
 	s.key = s.cmd.Arg("key", "Key path. Format is <objectType>/<objectInstance>/<key>.)").Required().String()
 	s.editor = s.cmd.Flag("editor", "editor to execute").Envar("EDITOR").Default("/usr/bin/vi").String()
+	s.noTerminalSetup = s.cmd.Flag("no-terminal-setup", "do not set terminal i/o to edit the credential.").Short('T').Bool()
 	s.common = common
 }
 
@@ -90,13 +92,15 @@ func (s *secretsEdit) doEdit() {
 	tmpFile.Close()
 
 	cmd := exec.Command(*s.editor, fileName)
-	if !terminal.IsTerminal(int(os.Stdin.Fd())) {
-		gotrace.Error("Unable to edit %s. Not a terminal. Exiting.", *s.key)
-		return
+	if !*s.noTerminalSetup {
+		if !terminal.IsTerminal(int(os.Stdin.Fd())) {
+			gotrace.Error("Unable to edit %s. Not a terminal. Exiting.", *s.key)
+			return
+		}
+		cmd.Stdin = os.Stdin
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
 	}
-	cmd.Stdin = os.Stdin
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
 	err = cmd.Run()
 
 	if err != nil {

